nixos/modules/system/boot/networkd.nix

```diff
@@ -3111,8 +3090,38 @@ in
         };
       };
 
+    })
+  ];
+
+  stage2Config = let
+    cfg = config.systemd.network;
+    unitFiles = mkUnitFiles "" cfg;
+  in mkMerge [
+    (commonConfig config)
+
+    { environment.etc = unitFiles; }
+
+    (mkIf config.systemd.network.enable {
+
+      users.users.systemd-network.group = "systemd-network";
+
+      systemd.additionalUpstreamSystemUnits = [
+        "systemd-networkd-wait-online.service"
+        "systemd-networkd.service"
+        "systemd-networkd.socket"
+      ];
+
       environment.etc."systemd/networkd.conf" = renderConfig cfg.config;
 
+      systemd.services.systemd-networkd = {
+        wantedBy = [ "multi-user.target" ];
+        restartTriggers = map (x: x.source) (attrValues unitFiles) ++ [
+          config.environment.etc."systemd/networkd.conf".source
+        ];
+      };
+
+      systemd.services.systemd-networkd.aliases = [ "dbus-org.freedesktop.network1.service" ];
```

Comment:
Was there a reason this `aliases` was moved out of the attrset right above it? I mean it should work either way I think, but wasn't sure if there was some motivation to do this

Reply:
Oh, I guess that's just a byproduct of organically evolving this over a year :P I should move that.